File: backend/learning/Learning/Infrastructure/Messaging/Events/EducatorProfileUpdatedEvent.cs
namespace Learning.Infrastructure.Messaging.Events;

public sealed record EducatorProfileUpdatedEvent : BaseEvent
{
    public override string EventType { get; init; } = Constants.EducatorProfileUpdated;
    public string UserId { get; init; }
    public string FirstName { get; init; }
    public string LastName { get; init; }
    public string ImageUrl { get; init; }
}

File: backend/profile/src/main/java/com/example/profile/infrastructure/rabbitmq/events/EducatorProfileUpdatedEvent.java
package com.example.profile.infrastructure.rabbitmq.events;

import com.example.profile.infrastructure.rabbitmq.Constants;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@NoArgsConstructor
@ToString(callSuper = true)
public class EducatorProfileUpdatedEvent extends BaseEvent {
    private String userId;
    private String firstName;
    private String lastName;
    private String imageUrl;

    public EducatorProfileUpdatedEvent(String userId, String firstName, String lastName, String imageUrl) {
        super();
        setEventType(Constants.EDUCATOR_PROFILE_UPDATED);
        this.userId = userId;
        this.firstName = firstName;
        this.lastName = lastName;
        this.imageUrl = imageUrl;
    }
}